cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -2499,6 +2509,11 @@ optionSolverFlags showOrParseArgs getmbj setmbj getrg setrg getcc setcc
             (toFlag `fmap` parsec))
          (flagToList . fmap prettyShow))
 
+  , option [] ["require-artifacts"]
+      "Reject installed dependency package options that are missing required build artifacts (default)."
```

Comment:
As you've added a new flag, can you add documentation of it to the appropriate place in the manual? (I hope there is such a place!)

Reply:
It's added.  (And to answer your question, it's more the latter.)